dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkApplicationListener.java

```diff
@@ -84,7 +85,7 @@ else if (event.getType() == RequestEvent.Type.RESP_FILTERS_START) {
                         commitTransaction();
                     } catch (Exception e) {
                         rollbackTransaction();
-                        this.<RuntimeException>rethrow(e);
+                        throw new MappableException(e);
```

Comment:
The original version casted every Exception to `RuntimeException`. Might this have caused the problem with Jersey not trying to map the specific exception with a registered exception mapper?